### PR TITLE
mon: monmap epoch can not to be 0

### PR DESCRIPTION
**Pull request by x11562**
Signed-off-by: Na Xie xie.na@h3c.com

### CODE REVIEW

File: src/mon/MonmapMonitor.cc [hunk 1]
```diff
@@ -200,6 +200,11 @@ bool MonmapMonitor::preprocess_command(MonOpRequestRef op)
       p = new MonMap;
       p->decode(bl);
     }
+    else {
```
Comment (tchaikov) [inline src/mon/MonmapMonitor.cc]: maybe we can just kill the check at https://github.com/ceph/ceph/pull/9650/files#diff-48d6f7c4cfa3781fd3f8ed5acd51b886R191.

Comment (x11562) [inline src/mon/MonmapMonitor.cc]: lgtm

**Comment by dachary** [comment]
jenkins test this please (jenkins hung)

**Comment by dachary** [comment]
jenkins test this please (asok)

**Comment by tchaikov** [comment]
@x11562 could you elaborate a little bit in the commit message explaining why "monmap epoch can not to be 0"?

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
I believe monmap epoch can be 0 for a brief moment during mkfs right after the initial quorum is formed but before the first paxos round with the initial monmap is completed.

**Comment by jecluis** [comment]
fwiw, @liewegas is right.

**Comment by tchaikov** [comment]
thanks guys.

**Pull request closed without merging**